Widgets - Validate

pannga opened the pull request:
Fixed some problems with Validate widget + fixed trackingId error

darekg11 reviewed (changes requested):
Does not work at all.
Set it up without amount field and widget ends up sending: `&amount=NaN` and it always comes back with `400 - Bad request`

EDIT: God damnit, nevermind. My bad.

pannga commented:
*tips fedora* thank you

pannga commented:
I've simplified this app state as @darekg11 requested.

I've also removed this `withCredentials` attribute which I've added here by mistake -> I won't be adding this in the final version (I've talked about this with @pawelrychlik)

I still wait for PR #47 to be merged in order for this PR to pass linting.

pannga commented:
@darekg11 I've merged those types and widget is ready for testing.

darekg11 commented on packages/react-widget/src/VoucherifyValidate.tsx:
```diff
@@ -1,13 +1,19 @@
 import * as React from 'react'
```
Remove amount input, use gift card, get 400 response but all of the controls are in green color indicating like it was a success while in reality user got 400

darekg11 commented on packages/sdk/src/types/ClientSide.ts:
```diff
@@ -9,9 +9,12 @@ export interface ClientSideValidateParams {
 	code?: string
```
Did you rebase your branch on top of master? I don't get why I see the same changes here which already have been applied in previous commit: https://github.com/voucherifyio/voucherify-js-sdk/commit/bf4e2ffdf018c4e6de9d256ea4a25e9b2d280cc6

darekg11 commented on packages/sdk/src/types/Validations.ts:
```diff
@@ -22,6 +22,9 @@ export interface ValidationsValidateVoucherParams {
 	gift?: {
```
Did you rebase your branch on top of master? I don't get why I see the same changes here which already have been applied in previous commit: bf4e2ff

darekg11 commented on packages/sdk/src/types/ClientSide.ts:
```diff
@@ -121,6 +142,23 @@ export interface VoucherifyRedeemInputs {
 	voucherifyTracking: string
 }
+
+export interface VoucherifyValidateInputsState {
```
Let's please move those types which are strictly related to widget to some new file in `packages/react-widget` dir. It does not make sense for someone who only downaloaded SDK package to see those types.

darekg11 reviewed (changes requested):
Small improvements required.
